include/wx/private/uilocale.h

```diff
@@ -0,0 +1,54 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/private/uilocale.h
+// Purpose:     wxUILocaleImpl class declaration
+// Author:      Vadim Zeitlin
+// Created:     2021-08-01
+// Copyright:   (c) 2021 Vadim Zeitlin <vadim@wxwidgets.org>
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_PRIVATE_UILOCALE_H_
+#define _WX_PRIVATE_UILOCALE_H_
+
+#include "wx/localedefs.h"
+#include "wx/string.h"
+
+// ----------------------------------------------------------------------------
+// wxUILocaleImpl provides the implementation of public wxUILocale functions
+// ----------------------------------------------------------------------------
+
+class wxUILocaleImpl
+{
+public:
+    // This function is implemented in platform-specific code and returns the
+    // object used by default, i.e. if wxUILocale::UseDefault() is not called.
+    // This object corresponds to the traditional "C" locale.
+    //
+    // It should never return NULL.
+    static wxUILocaleImpl* CreateStdC();
+
+    // Similarly, this one returns the object corresponding to the default user
+    // locale settings which is used if wxUILocale::UseDefault() was called.
+    // Under MSW and Mac it is the same as CreateStdC(), but it's different
+    // under Unix.
```

Comment:
Sorry, this comment is outdated, it was true for the original version of the code, but I've changed it since then to remain compatible with the current behaviour.